rootston/desktop.c

```diff
@@ -353,11 +356,17 @@ struct roots_desktop *desktop_create(struct roots_server *server,
 			&desktop->xwayland_surface);
 		desktop->xwayland_surface.notify = handle_xwayland_surface;
 
-		if (roots_xcursor_theme_load(desktop->xcursor_theme, 1)) {
-			wlr_log(L_ERROR, "Cannot load xwayland xcursor theme");
+		if (wlr_xcursor_manager_load(desktop->xcursor_manager, 1)) {
+			wlr_log(L_ERROR, "Cannot load XWayland XCursor theme");
+		}
+		struct wlr_xcursor *xcursor = wlr_xcursor_manager_get_xcursor(
+			desktop->xcursor_manager, ROOTS_XCURSOR_DEFAULT, 1);
+		if (xcursor != NULL) {
+			struct wlr_xcursor_image *image = xcursor->images[0];
+			wlr_xwayland_set_cursor(desktop->xwayland, image->buffer,
+				image->width, image->width, image->height, image->hotspot_x,
+				image->hotspot_y);
```

Comment:
I wonder if it makes sense to change the xwayland default cursor depending on which output the cursor is on.

Reply:
Well, there's not always just one cursor - and the XWayland cursor applies to all outputs, so I don't think this could work.

Reply:
Yeah it would only work in a one-cursor context but that's also the most common context we'll be running in

Reply:
I'd like to do this, by the way - spin it out into a separate ticket please.